packages/next/client/image.tsx

```diff
@@ -418,6 +418,11 @@ export default function Image({
         `Image with src "${src}" has both "priority" and "loading='lazy'" properties. Only one should be used.`
       )
     }
+    if (sizes && layout !== 'fill' && layout !== 'responsive') {
+      console.warn(
+        `Image with src "${src}" has "sizes" property but it will be ignored. Only use "sizes" with "layout='fill'" or "layout='responsive'".`
```

Comment:
```suggestion
        `Image with src "${src}" has "sizes" property but it will be ignored. "sizes" is only used with "layout='fill'" or "layout='responsive'".`
```